src/bridge-canister/src/runtime.rs

```diff
@@ -51,18 +51,24 @@ impl<Op: Operation> BridgeRuntime<Op> {
 
     /// Run the scheduled tasks.
     pub fn run(&mut self) {
-        if !self.state.borrow().collecting_logs {
-            self.state.borrow_mut().collecting_logs = true;
+        if let Some(collect_log_lock) = self.state.borrow_mut().collecting_logs.try_lock() {
             let task = scheduler::BridgeTask::Service(ServiceTask::CollectEvmLogs);
             let collect_logs = ScheduledTask::new(task);
             self.scheduler.append_task(collect_logs);
+
+            // Drop the lock
+            drop(collect_log_lock);
```

Comment:
I think the lock should be dropped when task execution finished, not when the task appended to the scheduler. 

Inside `ServiceTask::execute()`.